rosbag2_transport/test/rosbag2_transport/test_play.cpp

```diff
@@ -205,35 +208,60 @@ class RosBag2PlayQosOverrideTestFixture : public RosBag2PlayTestFixture
   RosBag2PlayQosOverrideTestFixture()
   : RosBag2PlayTestFixture()
   {
-    // Because this test only cares about compatibility (receiving any messages at all)
+  }
+
+  void initialize(const std::vector<rosbag2_transport::Rosbag2QoS> & offered_qos)
+  {
+    // Because these tests only cares about compatibility (receiving any messages at all)
     // We publish one more message than we expect to receive, to avoid caring about
     // shutdown edge behaviors that are not explicitly being tested here.
     const size_t num_msgs_to_publish = num_msgs_to_wait_for_ + 1;
-    topic_timestamps_ms_.reserve(num_msgs_to_publish);
+    messages_.reserve(num_msgs_to_publish);
     for (size_t i = 0; i < num_msgs_to_publish; i++) {
-      topic_timestamps_ms_.push_back(start_time_ms_ + message_spacing_ms_ * i);
+      const auto timestamp = start_time_ms_ + message_spacing_ms_ * i;
+      messages_.push_back(serialize_test_message(topic_name_, timestamp, basic_msg_));
     }
 
-    messages_.reserve(topic_timestamps_ms_.size());
-    for (const auto topic_timestamp : topic_timestamps_ms_) {
-      messages_.push_back(serialize_test_message(topic_name_, topic_timestamp, basic_msg_));
+    std::string serialized_offered_qos = "";
+    if (!offered_qos.empty()) {
+      YAML::Node offered_qos_yaml;
+      for (const auto & profile : offered_qos) {
+        offered_qos_yaml.push_back(profile);
+      }
+      serialized_offered_qos = YAML::Dump(offered_qos_yaml);
     }
+    topic_types_.push_back(
+      {topic_name_, msg_type_, "" /*serialization_format*/, serialized_offered_qos});
 
     auto prepared_mock_reader = std::make_unique<MockSequentialReader>();
     prepared_mock_reader->prepare(messages_, topic_types_);
     reader_ = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
   }
 
+  template<typename Duration>
+  void play_and_wait(Duration timeout, bool expect_timeout = false)
+  {
+    auto await_received_messages = sub_->spin_subscriptions();
+    Rosbag2Transport transport(reader_, writer_, info_);
```

Comment:
```suggestion
    Rosbag2Transport transport{reader_, writer_, info_};
```

Reply:
I thought this was only necessary for single-argument constructors

Reply:
IIRC: the benefit of brace-initialization is that it makes brace initialization unambiguous from function calls.

Reply:
you mean function declarations?

Reply:
oh I see; this constructer isn't actually potentially ambiguous here.
It would be if you did `auto transport = Rosbag2Transport(reader_, writer_, info_);`